tests/tests_pytorch/callbacks/test_device_stats_monitor.py

```diff
@@ -166,3 +168,45 @@ def test_device_stats_monitor_warning_when_psutil_not_available(monkeypatch, tmp
     # TODO: raise an exception from v1.9
     with pytest.warns(UserWarning, match="psutil` is not installed"):
         monitor.setup(trainer, Mock(), "fit")
+
+
+def test_device_stats_monitor_logs_for_different_stages(tmpdir):
+    """Test that metrics are logged for all stages that is training, testing and validation."""
+
+    model = BoringModel()
+    device_stats = DeviceStatsMonitor()
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_train_batches=7,
```

Comment:
1 epoch should be fine.
And it doesn't need 7 batches, 1 should be fine.
You should also set `limit_test_batches=1`

Reply:
I will change epochs and test batches for faster testing and thank you for all your suggestions 🙏.